pkg/providers/util/claim_extractor.go

```diff
@@ -95,8 +97,12 @@ func (c *claimExtractor) loadProfileClaims() (*simplejson.Json, error) {
 		return simplejson.New(), nil
 	}
 
-	claims, err := requests.New(c.profileURL.String()).
-		WithContext(c.ctx).
+	requestBuilder := requests.New(c.profileURL.String())
+	if client, ok := c.ctx.Value(oauth2.HTTPClient).(*http.Client); ok {
+		requestBuilder = requestBuilder.WithClient(client)
+	}
```

Comment:
As this block is used in three places already could we maybe abstract this block away into something like a `NewFromContext(ctx context.Context, profileUrl string)` or `WithClientFromContext(...)`?